### DISCUSSION TITLE
Dependency injection for lifespan

### DISCUSSION THREAD

**Discussion by sfc-gh-plis** [fastapi/fastapi#11427] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
N/A
```

### Description

I was hoping to inject pydantic settings into a lifespan function, similar to how you can do it now with endpoints.

### Operating System

macOS

### FastAPI Version

0.110.1

### Pydantic Version

2.6.4

### Python Version

3.10

### Additional Context

Right now best approach I found is to mimic it like so: 
```python
@asynccontextmanager
async def lifespan(app: FastAPI, settings: Settings = get_settings())
```

and in unit tests figure out a way to apply `functools.partial` equivalent for the `asynccontextmanager` and pass it to the `lifespan` parameter of the `FastAPI` constructor.

**Comment by jgould22** [comment]
In order to head off any XY problems is there any particular reason the usual [LRU and import method](https://fastapi.tiangolo.com/advanced/settings/?h=settings#creating-the-settings-only-once-with-lru_cache) will not work here?

Generally depends does not work outside of the route decorator.

**Reply by sfc-gh-plis** [reply]
Well, I guess it would be the same case why [it could not work for the routers](https://fastapi.tiangolo.com/advanced/settings/#settings-in-a-dependency) :)

**Reply by sfc-gh-plis** [reply]
I added my current workaround in the Additional Context section.
